aws-sdk-core/lib/aws-sdk-core/param_validator.rb

```diff
@@ -60,7 +60,7 @@ def structure(ref, values, errors, context)
     def list(ref, values, errors, context)
       # ensure the value is an array
       unless values.is_a?(Array)
-        errors << "expected #{context} to be an array"
+        errors << "expected #{context} to be an array, got value #{values} (class: #{values.class}) instead."
```

Comment:
I would suggest DRY'ing up the code by having a single re-usable string for the got value (class) instead messaging. You can create a const string at the top of the class like:

```ruby
GOT_INSTEAD = "got value %s (class: %s) instead."
```

Then you can simply append the following to each error message

```ruby
GOT_INSTEAD % [values.inspect, values.class.name]
```